### ISSUE TITLE
[macOS] Image attachments are not downloaded on click when using Manual attachment download mode

### ISSUE THREAD

**Issue by aism** [laurent22/joplin#11553]
### Operating system

macOS

### Joplin version

3.1.24

### Desktop version info

Joplin 3.1.24 (prod, darwin)

Client ID: 7425d85b87da42fbbfc7a8ce3f884647
Sync Version: 3
Profile Version: 47
Keychain Supported: Yes

Revision: d581264

Backup: 1.4.2
Inline tags: 1.3.0
Jarvis: 0.9.1
Link Graph UI: 1.5.0
Note Link System: 0.8.0
Note Tabs: 1.4.0
Rich Markdown: 0.15.1
Table Formatter Plugin: 1.2.1

### Current behaviour

In an attempt to fix the problem:
1. Closed Joplin
2. Deleted the local `~/.config/joplin-desktop` profile folder 
3. Moved settings and plugins into a pre-created `~/.config/joplin-desktop` folder:
```
settings.json
plugins/
```
4. Started Joplin and Re-configured plugin settings (as they aren't part of Joplin's `settings.json`)
5. Enabled sync with Dropbox and entered Master password
8. Waited for a sync to finish
9. Images are not loaded (seems expected in Manual mode):
How it looks like:
<img width="611" alt="SCR-20241222-uczl" src="https://github.com/user-attachments/assets/243c3efb-cdde-43dd-a529-8faa54f93d12" />

Same note from the other machine (where the attachment has been added locally):
<img width="695" alt="PastedGraphic-1" src="https://github.com/user-attachments/assets/86123e41-4b88-4586-80b1-e3cf561e96f5" />

10. Clicking on the unloaded image attachment in the Markdown editor mode. Getting an error:
<img width="285" alt="SCR-20241222-txvr" src="https://github.com/user-attachments/assets/ff081b49-6c07-499d-a5dd-8eed9db2a8f8" />

11. Switching to the WYSIWIG. Seeing an error, and cannot get back to the Markdown editor (need to select another note and switch there, and then get back to this note):
<img width="755" alt="SCR-20241222-txyl" src="https://github.com/user-attachments/assets/af1d2384-12a6-4f23-8f88-a316608ba7b1" />

The markdown is the same on both machines: `![b7c0e17a2ee3d200b2741b304ee3879f.png](:/ab486816ee5741a2924b7bebfe1d8350)`.

Searching by this ID in the `Tools > Note attachments` returns the attachment, but when clicking on the link nothing happens (it does open the image when doing it for the downloaded image):
<img width="1893" alt="SCR-20241222-udkf" src="https://github.com/user-attachments/assets/39dc863f-09d8-47a5-8273-ef1a2d2c7c14" />

The log from the debug mode restart is in the attachment.
[joplin_dev_console.log](https://github.com/user-attachments/files/18223349/joplin_dev_console.log)

Indeed, it contains the `net::ERR_FILE_NOT_FOUND` errors and indeed those files aren't downloaded to the profile folder.

P.S.: initially the problem was worded differently as I didn't notice that I have the Attachments synced in Manual mode and was confused by it. However, when troubleshooting it and trying to click around I discovered the behavior described here.

### Expected behaviour

Attachments should be loaded on-click and images displayed.

### Logs

Attached as a file.

**Comment by laurent22** [comment]
Did you recently change your sync target? This problem can happen in this situation

**Comment by aism** [comment]
@laurent22 No, I didn't - all my machines have been using Dropbox for years. The sequence was:
1. install Joplin on a new machine
2. supply it with settings (from another machine)
3. configure sync target (Dropbox) and then the master password

Although, I've just checked, and the `"sync.target": 7,` is part of the `settings.json`. Unfortunately, I don't remember what was at the time in UI and whether I had to select the sync target from scratch or just press a button to setup the integration b/w Joplin and Dropbox.

**Comment by aism** [comment]
Btw, I can reproduce this problem on my setup: if I just move back from "always" to "manual" then I'll have the same symptoms with it.
